python/taichi/main.py

```diff
@@ -69,7 +70,25 @@ def __init__(self, test_mode: bool = False):
 
         self.main_parser = parser
 
-        self._check_version()
+        # Check timestamp
+        os.makedirs(_ti_core.get_repo_dir(), exist_ok=True)
+        timestamp_path = os.path.join(_ti_core.get_repo_dir(), 'timestamp')
+        cur_date = datetime.date.today()
+        if os.path.exists(timestamp_path):
+            last_time = ''
+            with open(timestamp_path, 'r') as f:
+                last_time = f.readlines()[0].rstrip()
+            target_date = (cur_date -
+                           datetime.timedelta(days=14)).strftime('%Y-%m-%d')
+            if target_date > last_time:
+                with open(timestamp_path, 'w') as f:
+                    f.write(cur_date.strftime('%Y-%m-%d'))
+                    f.truncate()
+                self._check_version()
+        else:
+            with open(timestamp_path, 'w') as f:
```

Comment:
nit: instead of saving the cur_date, shall we just save (cur_date+14 days) in the file so that your just need to perform a simple comparison every time? ;)

Reply:
Make sense!